lua/nvim-tree.lua

```diff
@@ -17,6 +17,31 @@ local open = require "nvim-tree.actions.tree.open"
 local events = require "nvim-tree.events"
 local notify = require "nvim-tree.notify"
 
+---@class ParentNode
+---@field name string
+
+---@class BaseNode
+---@field absolute_path string
+---@field executable boolean
+---@field fs_stat uv_fs_t
+---@field git_status GitStatus|nil
+---@field hidden boolean
+---@field name string
+---@field parent DirNode
+---@field type string
+---@field watcher function|nil
+
+---@class DirNode: BaseNode
+---@field has_children boolean
+---@field group_next Node|nil
+---@field nodes Node[]
+---@field open boolean
+
+---@class FileNode: BaseNode
+---@field extension string
+
+---@alias Node ParentNode|DirNode|FileNode
```

Comment:
This is the `Node` type I managed to come up with, let me know if it's ok and most importantly where it should go. I put it here just to put it temporarily somewhere.

Reply:
> where it should go.

Where is indeed difficult. We could just create `nvim-tree/node.lua` just to hold the definitions.

When we eventually get to creating proper node classes they can live in there.

Reply:
> This is the Node type I managed to come up with, let me know if it's ok 

It's great! It contains all that we know about.

LLS can tell us if we're using an unknown field and we can add as we discover them.